### PR TITLE
docs(README): make types more explicit in examples

### PR DESCRIPTION
**Pull request by BrainCrumbz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,27 +18,32 @@ Based on [redux-saga-rxjs](https://github.com/salsita/redux-saga-rxjs) by Salsit
...
-  return iteration$ => iteration$
-    .filter(iter => iter.action.type === 'DECREMENT')
-    .map(() => {
-      return { type: 'INCREMENT'}
-    });
+export const incrementProvider = createSaga(function() {
+  return function incrementSaga(iteration$): Observable<Action> {
+    return iteration$
+      .filter(iter => iter.action.type === 'DECREMENT')
+      .map(() => {
```
Comment (MikeRyanDev) [inline README.md:32]: I think instead it should be:

``` ts
.map<Action>(...)
```

Comment (BrainCrumbz) [inline README.md:32]: We're having issues with all RxJs operators in VS 2015, although they compile fine (even without the generic type added): 

map-squiggle

so I can just follow your suggestion, "blindly", so to say.

PS we do `import 'rxjs/add/operator/map';`

Comment (MikeRyanDev) [inline README.md:32]: You will need to import the map operator for the typings to work:

``` ts
import 'rxjs/add/operator/map';
```

Comment (BrainCrumbz) [inline README.md:32]: Just edited my reply, I guess too late :smile: ... we have that import already. Everytime we save the file, red squiggles disappear for a moment, then they're back again. We only see those on all RxJs operators and observable static methods, although we import them:

``` ts
import 'rxjs/add/observable/of';
import 'rxjs/add/operator/catch';
```

We still have RxJs type definitions inside `typings` folder, although in more recent version the library itself in `node_modules` comes with its own type definitions ... don't know if that is a problem for VS.... we'll try uninstalling those typings.

Comment (BrainCrumbz) [inline README.md:32]: Hello there. In the team we tried working on our code with Visual Studio Code: things look better there, no spurious red squiggles. 

`map` seems fine without the type specifier:

map-no-type

and adding the type specifier seems to add nothing:

map-with-type

I think it makes sense, when I look at `map` type definition:

``` js
declare module '../../Observable' {
    interface Observable<T> {
        map: MapSignature<T>;
    }
}
// ... in imported file:
export interface MapSignature<T> {
    <R>(project: (value: T, index: number) => R, thisArg?: any): Observable<R>;
}
```

as it is the Observable that "receives" (so to say) a type. Once that is set, the other "loose end" (`<R>`) gets fixed by the type returned by projection function.

What do you think about it? If you think it's needed anyway, no probs: we can add that so to move on with the PR.

**Pull request closed without merging**